gnucash/report/standard-reports/transaction.scm

```diff
@@ -972,11 +1022,35 @@ disable the substring filter. This filter is case-sensitive.")
     "d"  (N_ "Display the memo?") #t
     #f
     (lambda (x) (gnc-option-db-set-option-selectable-by-name
-		 gnc:*transaction-report-options*
-		 gnc:pagename-display
-		 (N_ "Notes")
-		 x))))
-
+                 gnc:*transaction-report-options*
+                 gnc:pagename-display
+                 (N_ "Notes")
+                 x))))
+  
+  ;; Ditto for Account Name #t -> Use Full Account Name is selectable
+  (gnc:register-trep-option
+   (gnc:make-complex-boolean-option
+    gnc:pagename-display (N_ "Account Name")
+    "e"  (N_ "Display the account name?") #t
+    #f
+    (lambda (x) (gnc-option-db-set-option-selectable-by-name
+                 gnc:*transaction-report-options*
+                 gnc:pagename-display
+                 (N_ "Use Full Account Name")
+                 x))))
+  
+  ;; Ditto for Other Account Name #t -> Use Full Other Account Name is selectable
+  (gnc:register-trep-option
+   (gnc:make-complex-boolean-option
+    gnc:pagename-display (N_ "Other Account Name")
+    "h"  (N_ "Display the other account name? (if this is a split transaction, this parameter is guessed).") #f
```

Comment:
"h" is already in use as position indicator for the "Amount of detail to display per transaction." option. On my system this results in a wrong ordering of these two options. You can fix this by choosing "hx" for the "Other Account Name" option.